### PR TITLE
feat: cleanup indexer transaction.ts issue#1221

### PR DESCRIPTION
**Pull request by dznes**
Reorganize and cleanup indexer transaction.ts

Time spent on this PR: 8 hours

Resolves: #1221 

# Pull Request type
Reorganize and cleanup indexer transaction.ts from issue #1221 

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [X] Documentation content changes
- [X] Testing

#Detailed Description
The following changes have been made to the indexer transaction.ts file:

Comments and Documentation: Added detailed comments and documentation to explain the purpose of each function, its parameters, and its return values.
Error Handling: Improved error handling by adding clear and descriptive error messages.
Consistent Formatting: Ensured consistent formatting with proper indentation and spacing.
Descriptive Variable Names: Used meaningful and descriptive variable names for better readability.
Code Organization: Grouped related functions together logically and organized the code for better structure.
Simplified Logic: Refactored complex logic into smaller, more manageable functions.

Tests:
Added for toEthTx, chainId, setYParityFlag, setFlagRunOutOfResources and typedTransactionToEthTx functions

# What is the new behavior?
None

# Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by dznes** [comment]
@tcoratger

File: indexer/src/types/transaction.ts [hunk 1]
```diff
@@ -67,96 +71,181 @@ export function toEthTx({
...
+}
+
+/**
+ * Adds the yParity field to the transaction result if it is an EIP-1559 or EIP-2930 transaction.
+ *
+ * @param typedTransaction - The typed transaction object.
+ * @param jsonTx - The JSON representation of the transaction.
+ * @param result - The transaction result object in Ethereum format.
+ */
+function setYParityFlag(
```
Comment (tcoratger) [inline indexer/src/types/transaction.ts:108]:
```suggestion
function flagYParity(
```

**Comment by tcoratger** [comment]
Small nit otherwise looks good to me, pending @greged93 @eugypalu

File: indexer/src/types/transaction.ts (hunk 1 above)
Comment (greged93) [reply indexer/src/types/transaction.ts:108]: I would keep the set here

File: indexer/src/types/transaction.ts [hunk 2]
```diff
@@ -67,96 +71,181 @@ export function toEthTx({
...
+}
+
+/**
+ * Adds the isRunOutOfResources flag to the transaction result if the transaction
+ * was reverted due to running out of resources.
+ *
+ * @param receipt - The transaction receipt object.
+ * @param result - The transaction result object in Ethereum format.
+ */
+function flagRunOutOfResources(
```
Comment (greged93) [inline indexer/src/types/transaction.ts:130]: same here, I think the `set` is more explicit @tcoratger what do you think?

Comment (tcoratger) [reply indexer/src/types/transaction.ts:130]: yes let us put the `set` prefix in both cases

**Review by eugypalu — DISMISSED** [review]
Looks good to me

**Comment by dznes** [comment]
I can see one of the Workflow - Pull Request / Rust tests / test-deno (pull_request) failed, is there something i need to change in the code?
I ran the script: "KAKAROT_ADDRESS=0x1 deno test --allow-env" and all test passed before commiting the code.

**Comment by greged93** [comment]
I'm looking into it, I think there is a small issue with the CI

**Comment by greged93** [comment]
@dznes it seems like you don't have enough test coverage in your PR, can you add some tests for the functions you added?

**Comment by dznes** [comment]
@greged93 i've made the tests for the functions and now its branch coverage is 70.1% and line coverage 87.2%.

**Comment by dznes** [comment]
hello, is there any changes needed for the PR to be merged?

**Comment by greged93** [comment]
hey @dznes we are just waiting on PR #1251 in order to have more coverage on the main function to be sure we don't break anything with any changes.

**Comment by greged93** [comment]
hey @dznes, can you just fix the conflicts?

**Review by tcoratger — APPROVED** [review]

**Pull request merged**